ray_julia_jll/gen/make.jl

```diff
@@ -0,0 +1,171 @@
+    end
+end
+
+function artifact_checksums(tarball::AbstractString)
+    return open(tarball, "r") do io
+        artifact_checksums(io)
+    end
+end
+
+# Compute the Artifact.toml `git-tree-sha1`.
+function tree_hash_sha1(tarball_path)
+    return open(GzipDecompressorStream, tarball_path, "r") do tar
+        SHA1(Tar.tree_hash(tar))
+    end
+end
+
+# Compute the Artifact.toml `sha256` from the compressed archive.
+function sha256sum(tarball_path)
+    return open(tarball_path, "r") do tar
+        bytes2hex(sha256(tar))
+    end
+end
+
+function remote_url(repo_root::AbstractString, name::AbstractString="origin")
+    return LibGit2.with(LibGit2.GitRepo(repo_root)) do repo
+        LibGit2.with(LibGit2.lookup_remote(repo, name)) do remote
+            LibGit2.url(remote)
+        end
+    end
+end
+
+function upload_to_github_release(
+    archive_path::AbstractString,
+    archive_url::AbstractString,
+    commit;
+    kwargs...
+)
+    return upload_to_github_release(archive_path, parse(URI, artifact_url), commit; kwargs...)
+end
+
+# TODO: Does this work properly with directories?
+function upload_to_github_release(archive_path::AbstractString, archive_uri::URI, commit; kwargs...)
+    # uri = parse(URI, artifact_url)
+    if archive_uri.host != "github.com"
+        throw(ArgumentError("Artifact URL is not for github.com: $(archive_uri)"))
+    end
+
+    m = match(GH_RELEASE_ASSET_PATH_REGEX, archive_uri.path)
+    if m === nothing
+        throw(ArgumentError(
+            "Artifact URL is not a GitHub release asset path: $(archive_uri)"
+        ))
+    end
+
+    # The `ghr` utility uses the local file name for the release asset. In order to have
+    # have the asset match the specified URL we'll temporarily rename the file.
+    org_archive_name = nothing
+    if basename(archive_path) != m[:file_name]
+        org_archive_name = basename(archive_path)
+        archive_path = mv(archive_path, joinpath(dirname(archive_path), m[:file_name]))
+    end
+
+    upload_to_github_release(m[:owner], m[:repo_name], commit, m[:tag], archive_path; kwargs...)
+
+    # Rename the archive back to the original name
+    if org_archive_name !== nothing
+        mv(archive_path, joinpath(dirname(archive_path), org_archive_name))
+    end
+
+    return nothing
+end
+
+
+function upload_to_github_release(owner, repo_name, commit, tag, path; token=ENV["GITHUB_TOKEN"])
+    # Based on: https://github.com/JuliaPackaging/BinaryBuilder.jl/blob/d40ec617d131a1787851559ef1a9f04efce19f90/src/AutoBuild.jl#L487
+    # TODO: Passing in a directory path uploads multiple assets
+    # TODO: Would be nice to perform parallel uploads
+    cmd = ```
+        $(ghr()) \
+        -owner $owner \
+        -repository $repo_name \
+        -commitish $commit \
+        -token $token \
+        $tag $path
+    ```
+
+    run(cmd)
+end
+
+if abspath(PROGRAM_FILE) == @__FILE__
+
+    repo_path = abspath(joinpath(@__DIR__, "..", ".."))
+    pkg_url = remote_url(repo_path)
```

Comment:
`pkg_url` I'm getting back is `"git@github.com:beacon-biosignals/ray.jl"` which is failing the checks in `upload_to_github_release`

The full `artifact_url` is `"git@github.com:beacon-biosignals/ray.jl/releases/download/v0.1.0/ray_julia.v0.1.0.aarch64-apple-darwin-libgfortran5-cxx11-julia_version+1.9.2.tar.gz"` which, when parsed as a URI has `achive_uri.host == ""`.

Secondly the `archive_ur.path` doesn't have the leading `/` so the regex check is also failing.

Reply:
You probably need to be using an HTTPS URL in order for `upload_to_github_release` to work instead of this SCP-style syntax. I would recommend to just convert the SCP-style syntax to an HTTP URL. There's a regex I wrote for the LibGit2 library which you can use to parse both: https://github.com/JuliaLang/julia/blob/572fa5055ecb709edd3d227a10b46549f82b72b2/stdlib/LibGit2/src/utils.jl#L9C7-L9C16